File: AalsiTrader/Utilities/AgentDefinitions.swift
import SwiftUI

struct AgentDefinition {
    let id: String
    let name: String
    let greek: String
    let role: String
    let color: Color
    let defaultStatus: String
}

enum AgentDefinitions {
    static let all: [AgentDefinition] = [
        AgentDefinition(id: "alpha", name: "Professor", greek: "α", role: "Research Agent", color: .alphaRed, defaultStatus: "active"),
        AgentDefinition(id: "beta", name: "Techno-Kid", greek: "β", role: "Technical Analyst", color: .betaTeal, defaultStatus: "active"),
        AgentDefinition(id: "gamma", name: "Risko-Frisco", greek: "γ", role: "Risk Manager", color: .gammaPurple, defaultStatus: "active"),
        AgentDefinition(id: "sigma", name: "Prime", greek: "Σ", role: "Trade Hunter", color: .sigmaGreen, defaultStatus: "active"),
        AgentDefinition(id: "theta", name: "Macro", greek: "θ", role: "Macro Watcher", color: .thetaOrange, defaultStatus: "sleeping"),
        AgentDefinition(id: "delta", name: "Booky", greek: "δ", role: "Trade Journal", color: .deltaBlue, defaultStatus: "sleeping"),
    ]

    static func definition(for id: String) -> AgentDefinition? {
        all.first { $0.id == id }
    }

    static func color(for agentId: String) -> Color {
        definition(for: agentId)?.color ?? .gray
    }

    static func greek(for agentId: String) -> String {
        definition(for: agentId)?.greek ?? "?"
    }

    static func name(for agentId: String) -> String {
        definition(for: agentId)?.name ?? agentId
    }
}

File: AalsiTraderAndroid/app/src/main/java/com/aalsitrader/android/util/AgentDefinitions.kt
package com.aalsitrader.android.util

import androidx.compose.ui.graphics.Color
import com.aalsitrader.android.model.AgentStatus
import com.aalsitrader.android.ui.theme.*

data class AgentDefinition(
    val id: String,
    val name: String,
    val greek: String,
    val role: String,
    val color: Color,
    val defaultStatus: AgentStatus,
)

object AgentDefinitions {
    val all = listOf(
        AgentDefinition("alpha", "Professor", "\u03B1", "Research Agent", AlphaRed, AgentStatus.active),
        AgentDefinition("beta", "Techno-Kid", "\u03B2", "Technical Analyst", BetaTeal, AgentStatus.active),
        AgentDefinition("gamma", "Risko-Frisco", "\u03B3", "Risk Manager", GammaPurple, AgentStatus.active),
        AgentDefinition("sigma", "Prime", "\u03A3", "Trade Hunter", SigmaGreen, AgentStatus.active),
        AgentDefinition("theta", "Macro", "\u03B8", "Macro Watcher", ThetaOrange, AgentStatus.sleeping),
        AgentDefinition("delta", "Booky", "\u03B4", "Trade Journal", DeltaBlue, AgentStatus.sleeping),
    )

    fun colorForAgent(agentId: String): Color {
        return all.find { it.id == agentId }?.color ?: TextSecondary
    }

    fun colorForHex(hex: String): Color {
        return when (hex.lowercase()) {
            "#ff6b6b", "ff6b6b" -> AlphaRed
            "#4ecdc4", "4ecdc4" -> BetaTeal
            "#a855f7", "a855f7" -> GammaPurple
            "#10b981", "10b981" -> SigmaGreen
            "#f97316", "f97316" -> ThetaOrange
            "#3b82f6", "3b82f6" -> DeltaBlue
            else -> TextSecondary
        }
    }
}
